src/data-access/MwApiLexemeCreator.ts

```diff
@@ -37,10 +45,27 @@ export default class MwApiLexemeCreator implements LexemeCreator {
 				language: lexemeLanguageItemId,
 				lexicalCategory: lexicalCategoryItemId,
 			} ),
+			errorformat: 'html',
+			formatversion: 2,
 		} );
 		const response = await this.api.postWithEditToken( params )
-			.catch( () => {
-				return Promise.reject( [ { type: 'unknown' } ] );
+			// jQuery promises can resolve and reject with multiple values;
+			// the third argument is either missing or the result
+			.catch( ( code: string, _?: unknown, result?: ApiResult ): Promise<never> => {
+				let errors: SubmitError[];
+				try {
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					errors = result!.errors.map( ( apiError ) => {
```

Comment:
Can't we just check if `results`  is an object with an `errors` key that holds an array with an `if` condition?
That way we would not need neither the try-catch nor the non-null-assertion here.